### ISSUE TITLE
hello 

### ISSUE THREAD

**Issue by alfonsoYuan** [munki/macadmin-scripts#24]
Traceback (most recent call last):
  File "installinstallmacos.py", line 593, in <module>
    main()
  File "installinstallmacos.py", line 509, in main
    catalog, args.workdir, ignore_cache=args.ignore_cache)
  File "installinstallmacos.py", line 402, in os_installer_product_info
    product_info[product_key] = parse_server_metadata(filename)
  File "installinstallmacos.py", line 278, in parse_server_metadata
    md_plist = read_plist(filename)
  File "installinstallmacos.py", line 75, in read_plist
    with open(filepath, "rb") as fileobj:
TypeError: coercing to Unicode: need string or buffer, NoneType found

**Comment by alfonsoYuan** [comment]
thanks

**Comment by gregneagle** [comment]
Could you provide a little more context here? I've not seen this, nor have I had any other reports from the large number of other people running this, so we'll need more context to understand what's happening here.

Are you running this under Python2 or Python3?
What options are you providing to the script?
Does it happen before the script asks you to choose a version to download?
If it happens _after_, which version did you choose? 
(etc)

In other words it would be far more helpful to see a transcript of your attempt to run this tool, ending up with this traceback so there's a better possibility of replicating the issue.

**Comment by gregneagle** [comment]
My best guess, though, is that one or more of the downloaded files in content/ are incomplete or corrupt. I'd remove that directory and try again.

**Comment by alfonsoYuan** [comment]
Thank you for solving it.
